### PR TITLE
[java] adds more tests for java support event class

### PR DESCRIPTION
**Pull request by dev-velo**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

extension of https://github.com/SeleniumHQ/selenium/pull/12578

Added tests and fixed a small ordering issue in WebDriverListener, this puts it more int line with the rest of the method params
### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by titusfortner** [comment]
I haven't run the tests, yet, but I think you need to run `./scripts/format.sh` on this.

**Comment by dev-velo** [comment]
@titusfortner sounds good, got other pr's out so Ill run the same format script on them as well

File: java/test/org/openqa/selenium/support/events/EventFiringDecoratorTest.java [hunk 1]
```diff
@@ -261,11 +712,9 @@ public void afterDismiss(Alert alert) {
                 "afterAnyCall switchTo",
                 "beforeAnyCall alert",
                 "afterAnyCall alert",
-                "beforeAnyCall dismiss",
```
Comment (titusfortner) [inline java/test/org/openqa/selenium/support/events/EventFiringDecoratorTest.java]: I don't think this should be removed?

File: java/test/org/openqa/selenium/support/events/EventFiringDecoratorTest.java [hunk 2]
```diff
@@ -261,11 +712,9 @@ public void afterDismiss(Alert alert) {
...
                 "beforeAnyAlertCall dismiss",
                 "beforeDismiss",
                 "afterDismiss",
-                "afterAnyAlertCall dismiss",
```
Comment (titusfortner) [inline java/test/org/openqa/selenium/support/events/EventFiringDecoratorTest.java]: This too looks like it should still be there

**Comment by dev-velo** [comment]
@titusfortner whoops good catch, missed that on the commit. Added both back and now and it passes locally
image

**Comment by titusfortner** [comment]
@RevealOscar can you rebase the changes after I merged the other PR? Thanks.

**Comment by dev-velo** [comment]
oh ew, I mightve messed something up when doing it via intellij. One sec Ill fix it

**Comment by titusfortner** [comment]
Yeah when we merge on GitHub then try to rebase locally it's always a pain. I wish the default on GH was to rebase. Anyway, I rebased and am re-running will merge when tests pass. Thanks.

**Pull request merged**